thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftSerializationFormats.java

```diff
@@ -55,14 +63,44 @@
      */
     public static final SerializationFormat TEXT_NAMED_ENUM = SerializationFormat.of("ttext-named-enum");
 
-    private static final Set<SerializationFormat> THRIFT_FORMATS =
-            ImmutableSet.of(BINARY, COMPACT, JSON, TEXT, TEXT_NAMED_ENUM);
+    /**
+     * A way to lookup the related {@link TProtocolFactory} from a {@link SerializationFormat}.
+     * Entries are provided via registered SPI {@link ThriftProtocolFactoryProvider} implementations.
+     */
+    private static final Map<SerializationFormat, TProtocolFactory> knownProtocolFactories;
+
+    static {
+        final List<ThriftProtocolFactoryProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(ThriftProtocolFactoryProvider.class,
+                                   ThriftProtocolFactoryProvider.class.getClassLoader()));
+        knownProtocolFactories = providers
+                .stream()
+                .map(ThriftProtocolFactoryProvider::entries)
+                .flatMap(Set::stream)
+                .collect(toImmutableMap(Entry::getSerializationFormat,
+                                        Entry::getTProtocolFactory));
+    }
+
+    /**
+     * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
+     *
+     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not a
+     *         known Thrift serialization format
+     */
+    public static TProtocolFactory get(SerializationFormat serializationFormat) {
```

Comment:
How about renaming to `tProtocolFactory` instead of `get`?

Reply:
I wanted to keep the same name as in `ThriftProtocolFactories` and I cannot rename as I should keep binary compatibility.

Reply:
It's deprecated and I think it's okay to use a new name.
It was `ThriftProtocolFactories.get(serializationFormat)` so it's obvious that the return value will be a `TProtocolFactory`.
But now it is `ThriftSerializationFormats` so I thought using the specific name would be better in this case. But not so strong on this. It's your choice. 😄

Reply:
fair enough. let me rename the new method then